### PR TITLE
Add run actor via API article

### PR DESCRIPTION
**Pull request by m-murasovs**

### CODE REVIEW

**Review by metalwarrior665 — APPROVED** [review]
Great, let's do this. I will later do PR to update the section with Sync run since we have sync runs with datasets now.

**Comment by metalwarrior665** [comment]
@mhamas Thanks a lot for the review!

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+- [Synchronously](#synchronous-flow) – Runs shorter than 5 minutes.
+- [Asynchronously](#asynchronous-flow) – Runs longer than 5 minutes.
+
+## [](#synchronous-flow) Synchronous flow
+
+If each of your runs is shorter than 5 minutes, you can use a single [synchronous endpoint](https://usergrid.apache.org/docs/introduction/async-vs-sync.html#synchronous). The connection is held for up to 5 minutes.
+
+If your run exceeds this time limit, the response will be a run object containing information about the run and the status `RUNNING`. If that happens, you need to [wait for the run to finish](#wait-for-the-run-to-finish).
+
+### [](#synchronous-runs-with-key-value-store-output) Synchronous runs with key-value store output
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:26]: Well, ideally the user will realize that he needs to go with async path in that case. Not sure about adding a code example there, don't want to make this article overly complicated. @m-murasovs What is your call? :)

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+[Key-value stores]({{@link storage/key_value_store.md}}) are useful for storing files like images, HTML snapshots or JSON data. The Apify API provides **run-sync** endpoints for
+[actors](/api/v2#/reference/actors/run-actor-synchronously/with-input)
+and [tasks](/api/v2#/reference/actor-tasks/run-task-synchronously/run-task-synchronously). These allow you to run a specific task and receive the output. By default, they return the `OUTPUT` record from the default key-value store.
+
+### [](#synchronous-runs-with-dataset-output) Synchronous runs with dataset output
+
+Most actor runs will store their data in the default [dataset]({{@link storage/key_value_store.md}}). The Apify API provides **run-sync-get-dataset-items** endpoints for
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:34]: @m-murasovs Will you please fit this? Thx

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+[actors](/api/v2#/reference/actors/run-actor-synchronously-and-get-dataset-items/run-actor-synchronously-with-input-and-get-dataset-items)
+and [tasks](/api/v2#/reference/actor-tasks/run-task-synchronously-and-get-dataset-items/run-task-synchronously-and-get-dataset-items-(post)). These endpoints allow you to run an actor and receive the items from the default dataset.
+
+A simple example of calling a task and logging the dataset items in Node.js.
+
+```javascript
+// Use your favourite HTTP client
+const got = require('got');
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:42]: We use `got` for `Apify.utils.requestAsBrowser`. I forgot about `@apify/http-request`, we should use that instead :)

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    method: 'POST',
+    json: {
+        queries: 'web scraping\nweb crawling'
+    },
+    responseType: 'json',
+});
+
+const items = response.body;
+
+// Log each organic result for both queries
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:61]: Non-paid results on Google, yeah we could just rename it to non paid

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+To run (or "call" in API language) an actor/task, you will need a few things:
+
+- Name or ID of the actor/task. The name is in the format `username~actorName` or `username~taskName`.
+
+- Your [API token](/api/v2#/introduction/authentication). You can find it on the Integrations page in the Apify [app](https://my.apify.com/account#/integrations) (make sure it does not leak anywhere!).
+
+- Possibly an input or other settings if you want to change the default values (e.g. memory or build).
+
+The template URL for a [POST request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) to run the actor looks like this:
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:98]: You are right, this should be on the top level. @m-murasovs Would you be able to restructure it a bit?

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+```text
+https://api.apify.com/v2/acts/ACTOR_NAME_OR_ID/runs/RUN_ID
+```
+
+Once you receive a `status` of `SUCCEEDED` or `FAILED`, you know the run has finished and you can cancel the interval and [collect the data](#collect-the-data).
+
+### [](#collect-the-data) Collect the data
+
+Unless you have used the [synchronous call](#synchronous-call) mentioned above, you will have to make one additional request to the API to retrieve the data.
```
[the remark this answers is not in the document]
Comment (metalwarrior665) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:201]: Y, missed this. @m-murasovs Please look at it

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-murasovs) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:26]: Maybe just rephrase to "If that happens, you will need to re-run the actor asynchronously and [wait for the run to finish](#wait-for-the-run-to-finish)." ?

The article is already pretty long, so let's try and avoid another code example.

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (m-murasovs) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:42]: I'd say it's pretty popular

Screenshot 2020-11-18 at 14 41 56

File: docs/tutorials/run_actor_and_retrieve_data_via_API.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (m-murasovs) [reply docs/tutorials/run_actor_and_retrieve_data_via_API.md:98]: Yep, I'll do it :)

**Pull request merged**